File: Part1_강의자료/Ch03_배열/문제별코드/1236_성지키기/1236.cpp
#include <algorithm>
#include <stdio.h>

using namespace std;

char castle[55][55];
int row_exist[55], col_exist[55];
int N, M;

int main()
{
    scanf(" %d %d", &N, &M);
    for (int i = 0; i < N; ++i)
        scanf(" %s", castle[i]);

    for (int i = 0; i < N; ++i)
    {
        for (int j = 0; j < M; ++j)
        {
            if (castle[i][j] == 'X')
            {
                row_exist[i] = 1;
                col_exist[j] = 1;
            }
        }
    }

    int row = N;
    for (int i = 0; i < N; ++i)
        row -= row_exist[i];
    
    int col = M;
    for (int i = 0; i < M; ++i)
        col -= col_exist[i];
    
    printf("%d\n", max(row, col));
    return 0;
}

File: Part1_강의자료/Ch03_배열/문제별코드/1236_성지키기/1236.py
N, M = map(int, input().split())
castle = [[0] * M for _ in range(N)]
row = 0
col = 0

for i in range(N):
    line = input()
    flag = False
    for j in range(M):
        castle[i][j] = line[j]
        if line[j] == 'X':
            flag = True
    if flag == False:
        col += 1

for i in range(M):
    if 'X' not in [castle[j][i] for j in range(N)]:
        row += 1

print(max(row, col))
